### PR TITLE
Update Rubocop to 0.52.1

### PR DESCRIPTION
**Pull request by timothysu**
rubocop `> 0.51.0` is needed to target ruby 2.5.0 and rubocop-rspec `> 1.21.0` is needed to support those versions of rubocop

prime: @jturkel

### CODE REVIEW

**Comment by timothysu** [comment]
@jturkel seems like a bunch of cops have been added in the meantime for both `rubocop` and `rubocop-rspec`.  Here's an example build of `content-flow-service`.  Any opinions on if any should be disabled, or if code should be refactored to conform?  https://circleci.com/gh/salsify/content-flow-service/1830

**Review by jturkel — COMMENTED** [review]
Are any changes required to our default Rubocop config for this upgrade? We typically rollout Rubocop upgrades to a few projects (e.g. webhooks, alerts, content flow) to shake out any issues before releasing a non-RC version of salsify_rubocop.

**Comment by jturkel** [comment]
Just saw your comment above. Do you have thoughts on which cops are valuable? Do the new cops have any kind of justification for why they're valuable?

**Comment by timothysu** [comment]
@jturkel ready for another pass

File: lib/rubocop/cop/salsify/rails_application_record.rb [hunk 1]
```diff
@@ -1,11 +1,7 @@
-# encoding: utf-8
-
-require 'rubocop'
-
 module RuboCop
   module Cop
     module Salsify
-      # Check that models subclass ApplicationRecord with Rails 5.0
+      # Check that models subclass ApplicationRecord with Rails 5.0.
```
Comment (jturkel) [inline lib/rubocop/cop/salsify/rails_application_record.rb]: Looks like this cop exists in rubocop now: https://github.com/bbatsov/rubocop/blob/10a7041d23bcd579821b378dd351aeead7c3f082/lib/rubocop/cop/rails/application_record.rb

File: lib/rubocop/cop/salsify/style_dig.rb [hunk 2]
```diff
@@ -1,5 +1,3 @@
-# encoding: utf-8
-
 module RuboCop
   module Cop
     module Salsify
```
Comment (jturkel) [inline lib/rubocop/cop/salsify/style_dig.rb]: While we're in here can we add a reference to https://github.com/bbatsov/rubocop/issues/5332?

**Review by jturkel — APPROVED** [review]
A few small comments but looks great. Thanks again for tackling this!

**Pull request merged**